File: 30/solution.rb
total = 0

(2..999999).each do |i|
  uberHyperCubeSum = i.to_s.split("").map { |char| char.to_i ** 5 }.reduce &:+
  total += i if i == uberHyperCubeSum
end

puts total

File: 30/solution.js
var total = 0;

for (var i = 2; i < 999999; i += 1) {
  var strumberray = i.toString().split('');
  var uberHyperCubeSum = 0;
  for (var index = 0; index < strumberray.length; index += 1) {
    uberHyperCubeSum += Math.pow(parseInt(strumberray[index], 10), 5);
  }
  if (uberHyperCubeSum === i) {
    total += i;
  }
}

console.log(total);
